aips/wallet-standard.md

```diff
@@ -0,0 +1,325 @@
+
+```ts
+getNetwork(): Promise<UserResponse<NetworkInfo>>;
+```
+
+`aptos:signAndSubmitTransaction` method to sign and submit a transaction using the current connected account in the wallet.
+
+```ts
+// `transaction: AnyRawTransaction` - a generated raw transaction created with Aptos’ TS SDK
+
+signAndSubmitTransaction(transaction: AnyRawTransaction): Promise<UserResponse<PendingTransactionResponse>>;
+```
+
+`aptos:signTransaction` for the current connected account in the wallet to sign a transaction using the wallet.
+
+```ts
+// `transaction: AnyRawTransaction` - a generated raw transaction created with Aptos’ TS SDK
+
+signTransaction(transaction: AnyRawTransaction):AccountAuthenticator
+```
+
+`aptos:signMessage` for the current connected account in the wallet to sign a message using the wallet.
+
+```ts
+// `message: AptosSignMessageInput` - a message to sign
+
+signMessage(message: AptosSignMessageInput):Promise<UserResponse<AptosSignMessageOutput>>;
+```
+
+`aptos:onAccountChange` event for the wallet to fire when an account has been changed in the wallet.
+
+```ts
+// `newAccount: AccountInfo` - The new connected account
+
+onAccountChange(newAccount: AccountInfo): Promise<void>
+```
+
+`aptos:onNetworkChange` event for the wallet to fire when the network has been changed in the wallet.
+
+```ts
+// `newNetwork: NetworkInfo` - The new wallet current network
+
+onNetworkChange(newNetwork: NetworkInfo):Promise<void>
+```
+
+`aptos:changeNetwork*` event for the dapp to send to the wallet to change the wallet’s current network
+
+```ts
+// `network:NetworkInfo` - - The network for the wallet to change to
+
+changeNetwork(network:NetworkInfo):Promise<UserResponse<{success: boolean,reason?: string}>>
+```
+
+`aptos:openInMobileApp*` a function that supports redirecting a user from a web browser on mobile to a native mobile app. The wallet plugin should add the location url a wallet should open the in-app browser at.
+
+```ts
+openInMobileApp(): void
+```
+
+Types
+
+> Note: `UserResponse` type is used for when a user rejects a rejectable request. For example, when user wants to connect but instead closes the window popup.
+
+```ts
+export interface UserApproval<TResponseArgs> {
+ status: 'approved'
+ args: TResponseArgs
+}
+
+export interface UserRejection {
+ status: 'rejected'
+}
+
+export type UserResponse<TResponseArgs> = UserApproval<TResponseArgs> | UserRejection;
+
+export interface AccountInfo = { account: Account, ansName?: string }
+
+export interface NetworkInfo {
+  name: Network
+  chainId: number
+  url?: string
+}
+
+export type AptosSignMessageInput = {
+  address?: boolean
+  application?: boolean
+  chainId?: boolean
+  message: string
+  nonce: string
+}
+
+export type AptosSignMessageOutput = {
+  address?: string
+  application?: string
+  chainId?: number
+  fullMessage: string
+  message: string
+  nonce: string
+  prefix: 'APTOS'
+  signature: Signature | Signature[]
+  bitmap?: Uint8Array
```

Comment:
I think this should be just `Signature`.
Both `MultiEd25519Signature` and `MultiKeySignature` already contain a bitmap field.

Reply:
@hardsetting do you mean we should remove the `bitmap` prop or change its type to `Signature`?

Reply:
I'm guessing `Signature` here is the abstract class / union type for _any_ signature
i.e. `Ed25519Signature | MultiEd25519Signature | AnySignature | MultiKeySignature`

since `MultiEd25519Signature` and `MultiKeySignature` already contain a field for `bitmap` and `signatures`, I would change this like such:

```suggestion
  signature: Signature
```